app/src/components/features/rules/RuleBuilder/Header/index.js

```diff
@@ -30,6 +36,19 @@ const Header = ({
     }`;
   };
 
+  useEffect(() => {
+    // groupwise rules are already created
+    if (Object.keys(groupwiseRulesToPopulate).length > 0) return;
```

Comment:
Can you please add a comment here which explains this `useEffect` for reference?

Reply:
yeah 👍